command/init.go

```diff
@@ -878,6 +880,14 @@ corresponding provider blocks in configuration, with the constraint strings
 suggested below.
 `
 
+const errDiscoveryServiceUnreachable = `
+[reset][bold][red]Registry service unreachable.[reset][red]
+
+This may indicate a network issue, or an issue with the requested Terraform Registry.
+
+Check https://status.hashicorp.com/ for registry status updates.
```

Comment:
This status page only really applies if the host we were accessing was `registry.terraform.io`, so maybe it'd be worth including the host as a field of the error message so we can mention which host "the requested Terraform Registry" is (there may be more than one used in a single configuration) and, perhaps, also omit this final line if it's a third-party one.

Reply:
I think it would be ok to switch on `if registry hostname == regsrc.PublicRegistryHost`, and otherwise either omit the line.

Either way we should ensure that the hostname attempted is displayed.

Reply:
Very true. There's more work planned to make this more useful, but I think this provides some help for the common path.